Neos.ContentRepository.Core/Classes/CommandHandler/CommandHandlingDependencies.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository.Core package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\CommandHandler;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphInterface;
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphReadModelInterface;
+use Neos\ContentRepository\Core\SharedModel\Exception\WorkspaceDoesNotExist;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamStatus;
+use Neos\ContentRepository\Core\SharedModel\Workspace\Workspace;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\EventStore\Model\Event\Version;
+
+/**
+ * Encapsulates the access to the read model which is used by the command handlers for soft constraints
+ *
+ * @internal
+ */
+final readonly class CommandHandlingDependencies
+{
+    public function __construct(
+        private ContentGraphReadModelInterface $contentGraphReadModel
+    ) {
+    }
+
+    public function getContentStreamVersion(ContentStreamId $contentStreamId): Version
+    {
+        $contentStream = $this->contentGraphReadModel->findContentStreamById($contentStreamId);
+        if ($contentStream === null) {
+            throw new \InvalidArgumentException(sprintf('Failed to find content stream with id "%s"', $contentStreamId->value), 1716902051);
+        }
+        return $contentStream->version;
+    }
+
+    public function contentStreamExists(ContentStreamId $contentStreamId): bool
+    {
+        $cs = $this->contentGraphReadModel->findContentStreamById($contentStreamId);
+        return $cs !== null && !$cs->removed;
```

Comment:
nitpick but according to our CGL:
```suggestion
        $contentStream = $this->contentGraphReadModel->findContentStreamById($contentStreamId);
        return $contentStream !== null && !$contentStream->removed;
```

And related to that: Wasn't the plan to get rid of the `removed` flag? (or is that part of a separate PR?)

Reply:
yeah this line will be fully eradicated via https://github.com/neos/neos-development-collection/pull/5297 so im not applying the cgl thing :D

Reply:
Sweet!